### ISSUE TITLE
Use IoC container to resolve dependencies

### ISSUE THREAD

**Issue by bartlannoeye** [ivaylokenov/MyTested.WebApi#254]
Next to testing routing, I'm looking to writing integration tests using this library in the form of your sample:

```
MyWebApi
    .Controller<WebApiController>()
    .Calling(c => c.SomeAction())
    .ShouldReturn()
    .Ok()
    .WithDefaultContentNegotiator()
    .WithResponseModelOfType<ResponseModel>();
```

My WebApi is set up with Autofac as IoC container and I'd like to re-use that for dependency resolution of all dependencies in the controller and the layers below (instead of having to redefine them all in the fluent API call above. Is this possible or something that could be added in the near future (maybe through PR) without major architectural changes?

Right now, this is how the Autofac container is coupled with ASP.NET WebApi:

```
var configuration = GlobalConfiguration.Configuration;

var resolver = new AutofacWebApiDependencyResolver(container);
configuration.DependencyResolver = resolver;
```

**Comment by ivaylokenov** [comment]
@bartlannoeye This should be so easy to add, I wonder why I have not implement it already. I will upload a new version with your requested feature in a day or two.

Because you said integration tests - you can write integration tests with AutoFact like this [sample](https://github.com/ivaylokenov/MyTested.WebApi/tree/master/samples/Source%20Control%20System%20API/Tests/SourceControlSystem.Api.Tests). However it runs full integration tests from request to response.

**Comment by bartlannoeye** [comment]
I'll have a look at that as well, thanks for the quick response.

**Comment by ivaylokenov** [comment]
@bartlannoeye Implemented in version 1.2.2. You can update the package. You can now provide global dependency resolver with the HTTP configuration. Documentation [HERE](https://github.com/ivaylokenov/MyTested.WebApi/tree/master/documentation#custom-dependency-resolver).

**Comment by bartlannoeye** [comment]
Awesome !
